swift-sdk/Internal/IterableEmbeddedManager.swift

```diff
@@ -101,19 +190,31 @@ class IterableEmbeddedManager: NSObject, IterableEmbeddedManagerProtocol {
     
     private func setMessages(_ processor: EmbeddedMessagingProcessor) {
         messages = processor.processedMessagesList()
+        cleanUpTrackedMessageIds(messages)
     }
     
+    private func cleanUpTrackedMessageIds(_ currentMessages: [Int: [IterableEmbeddedMessage]]) {
+        let currentUniqueKeys = Set(currentMessages.flatMap { placement, messages in
+            messages.map { "\(placement)-\($0.metadata.messageId)" }
+        })
+        trackedMessageIds = trackedMessageIds.intersection(currentUniqueKeys)
+    }
+
     private func trackNewlyRetrieved(_ processor: EmbeddedMessagingProcessor) {
-        for message in processor.newlyRetrievedMessages() {
-            IterableAPI.track(embeddedMessageReceived: message)
+        for (placementId, messages) in processor.newlyRetrievedMessages() {
+            for message in messages {
+                let messageId = message.metadata.messageId
+                let uniqueKey = "\(placementId)-\(messageId)"
```

Comment:
what is this unique key? 
placementId - messageId? what was the purpose of this?